client/widgets/CWindowWithArtifacts.cpp

```diff
@@ -338,10 +360,13 @@ std::optional<std::tuple<const CGHeroInstance*, const CArtifactInstance*>> CWind
 	// The state is possible when the hero has placed an artifact in the ArtifactPosition::TRANSITION_POS,
 	// and the previous artifact has not yet removed from the ArtifactPosition::TRANSITION_POS.
 	// This is a transitional state. Then return nullopt.
-	if(pickedCnt > 1)
+	if(std::accumulate(std::begin(pickedCnt), std::end(pickedCnt), 0, [](size_t accum, const std::map<const CGHeroInstance*, size_t>::value_type & value)
```

Comment:
Haven't checked, so might be wrong, but:
- you can use boost::range::accumulate for simplicity (if it exists)
- you can use `auto` instead of massive `std::map<const CGHeroInstance*, size_t>::value_type`

Reply:
> * you can use boost::range::accumulate for simplicity (if it exists)

I prefer to use std instead of boost where possible.

> * you can use `auto` instead of massive `std::map<const CGHeroInstance*, size_t>::value_type`

And thanks for the hint.